packages/cli/src/commands/proxy/utils.ts

```diff
@@ -0,0 +1,147 @@
+    target: targetUrl,
+    changeOrigin: true,
+    selfHandleResponse: hasResponseHandler,
+    ...conditional(
+      hasResponseHandler && {
+        on: {
+          proxyRes: onProxyResponse,
+        },
+      }
+    ),
+  });
+};
+
+export const createStaticFileProxy = (staticPath: string) => {
+  const index = 'index.html';
+  const indexContentType = 'text/html; charset=utf-8';
+  const noCache = 'no-cache, no-store, must-revalidate';
+  const maxAgeSevenDays = 'max-age=604_800_000';
+
+  return async (request: http.IncomingMessage, response: http.ServerResponse) => {
+    if (!request.url) {
+      response.writeHead(400).end();
+      return;
+    }
+
+    if (request.method === 'HEAD' || request.method === 'GET') {
+      const loadIndex = !isFileAssetPath(request.url);
+      const requestPath = path.join(staticPath, loadIndex ? index : request.url);
+      try {
+        const content = await fs.readFile(requestPath, 'utf8');
+        response.setHeader('cache-control', loadIndex ? noCache : maxAgeSevenDays);
+        response.setHeader('content-type', loadIndex ? indexContentType : getMimeType(request.url));
+        response.writeHead(200);
+        response.end(content);
+      } catch {
+        response.setHeader('content-type', getMimeType(request.url));
+        response.writeHead(404);
+        response.end();
+      }
+    }
+  };
+};
+
+/**
+ * Intercept the response from Logto endpoint and replace Logto endpoint URLs in the response with the
+ * proxy URL. The string replace happens in the following cases:
+ * - The response is a redirect response, and the `location` property in response header may contain Logto
+ *   endpoint URI.
+ * - The response body is JSON, which consists of properties such as `**_endpoint` and `redirectTo`. These
+ *   properties may contain Logto endpoint URI.
+ * - The response is HTML content that contains a form. The form action URL may contain Logto endpoint URI.
+ *
+ * Note: the `issuer` and `jwks_uri` properties in the `/oidc/.well-known` response should not be replaced,
+ * even they also contain the Logto endpoint URI.
+ */
+export const createLogtoResponseHandler = async ({
+  proxyResponse,
+  request,
+  response,
+  logtoEndpointUrl,
+  proxyUrl,
+  verbose,
+}: ProxyResponseHandler) => {
+  const { location } = proxyResponse.headers;
+  if (location) {
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    proxyResponse.headers.location = location.replace(logtoEndpointUrl.href, proxyUrl.href);
+  }
+
+  void responseInterceptor(async (responseBuffer, proxyResponse) => {
+    const responseBody = responseBuffer.toString();
+    if (verbose) {
+      consoleLog.info(`Response received: ${chalk.green(responseBody)}`);
+    }
+
+    if (proxyResponse.headers['content-type']?.includes('text/html')) {
+      return responseBody.replace(
+        `action="${logtoEndpointUrl.href}oidc/session/end/confirm"`,
+        `action="${proxyUrl.href}oidc/session/end/confirm"`
```

Comment:
maybe we don't need to match the path. use a regex will be good

Reply:
Updated with simplified string replace